types/framework/entities/chatMessage.d.ts

```diff
@@ -105,11 +105,7 @@ declare class ChatMessage extends Entity<ChatMessage.Data> {
   /* -------------------------------------------- */
 
   /** @override */
-  static create(data: DeepPartial<ChatMessage.CreateData>, options?: Entity.CreateOptions): Promise<ChatMessage | null>;
-  static create(
-    data: DeepPartial<ChatMessage.CreateData>[],
-    options?: Entity.CreateOptions
-  ): Promise<ChatMessage[] | null>;
+  static create: Entity.CreateFunction<ChatMessage>;
```

Comment:
ChatMessage.CreateData exists for a specific reason, so this overload won't work right when you pass the full entities in.

Reply:
Ah, because of `ChatMessage.CreateData`? Is it really different from `DeepPartial<ChatMessage['data']`?

Reply:
Ah, seems like it, it does preprocess the data apparently. I'll adjust it accordingly.

Reply:
Addressed it. It's using `ChatMessage.CreateData` again now.